metricbeat/beater/metricbeat.go

```diff
@@ -329,10 +285,22 @@ func (bt *Metricbeat) WithOtelFactoryWrapper(wrapper cfgfile.FactoryWrapper) {
 // result in undefined behavior.
 func (bt *Metricbeat) Stop() {
 	bt.stopOnce.Do(func() { close(bt.done) })
-
 }
 
 // Modules return a list of all configured modules.
 func (bt *Metricbeat) Modules() ([]*module.Wrapper, error) {
 	return module.ConfiguredModules(bt.registry, bt.config.Modules, bt.config.ConfigModules, bt.moduleOptions, bt.logger)
 }
+
+func MetricbeatAutodiscoverSetup(reg *autodiscover.Registry) error {
+	if err := initialization.Setup(reg); err != nil {
+		return fmt.Errorf("autodiscover initialization error: %w", err)
+	}
+	if err := hints.Setup(reg); err != nil {
```

Comment:
nit: I see that hnts is used to all filebeat/mettricbeat/heartbeat AutodiscoverSetup.
So maybe to consider to move it to main inintialise.Setup function as well?

Reply:
Unfortunately they aren't the same hints.

filebeat is `"github.com/elastic/beats/v7/filebeat/autodiscover/builder/hints"`
heartbeat is `	"github.com/elastic/beats/v7/heartbeat/autodiscover/builder/hints"`
metricbeat is `	"github.com/elastic/beats/v7/metricbeat/autodiscover/builder/hints"`